### PR TITLE
spec: Propose specification: Location

### LINKED ISSUE #2: Specification: Location
## Scenario and Use Case

Identifisere holdeplass via GPS

## Trigger

Bussen stopper eller åpner dør? Gateway sender GPS-posisjon. MT-Buss identifiserer holdeplass utfra posisjon.

## Achieved result

  a. Holdeplass funnet.
  b. Holdeplass ikke funnet utfra posisjon. Bruk nærmeste passerte holdeplass?
  c. Ikke mottatt GPS-posisjon. Sjåfør velger holdeplass manuelt?

### PR DESCRIPTION
**Pull request by mikaelbr**
# Proposal: Location

Retrieve location data periodically.

Fixes #2

## Context and known limitations

Specified as a subset of [Ruter Location](https://github.com/RuterNo/adt-doc/blob/master/docs/v2.6.0/json-schemas/location.json) but with local MQTT (no specifying operator and vehicle). Mainly support for lat, lon, and date.

Has no configuration like interval or format.

---

## Checklist

- [x] Includes documentation file with description, topics, and examples.
- [x] Includes JSON Schema
- [x] Includes passing and failing tests.

## Confirmed and verified by parties

- [x] MT Buss
- [ ] Consat
- [x] Fara

### CODE REVIEW

**Comment by mikaelbr** [comment]
- [x] Add fields about accuracy and ability to say something about quality (additional fields).
- [x] Information on movement.


---

Check with Consat on complete Ruter location JSON Schema

**Comment by mikaelbr** [comment]
Add note on standard interval. 1 per sec.

Max once

File: specifications/sensors/gnss/location.json [hunk 1]
```diff
@@ -0,0 +1,83 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "https://github.com/mrfylke/mtbuss-mqtt-validator/specifications/location.json",
```
Comment (Topguy) [inline specifications/sensors/gnss/location.json]: Full path har vel blitt  /specifications/sensors/gnss/location.json  ?

Comment (mikaelbr) [reply specifications/sensors/gnss/location.json]: Stemmer. Tenkte å restrukturere litt paths i dag for å samle dokumentasjon og schemas på samme sted.

**Comment by mikaelbr** [comment]
**Information**:
Ruter has an unmerged breaking change in how to handle Location in what they call adt v3. To be future proof and match Ruter I've updated this proposal to match Ruters _new_ Location specification.

Primary changes:
- Added required fields `traceId`, `messageNumber` and `eventTimestamp` (see description in schema)
- Removal of `latitudeDirection` and `longitudeDirection`
- Change topic to `/sensors/location`

**Comment by daniil-pastuhov** [comment]
:thumbsup: (MtBus)

**Comment by NilsAndreasT** [comment]
MT-Buss approves.

**Comment by mikaelbr** [comment]
@Topguy comments?

**Comment by Thorbear** [comment]
Approved by Fara, with the new adt v3 changes.

**Comment by mikaelbr** [comment]
Considering it approved and doing separate flow with Consat for now.

**Pull request merged**